### PR TITLE
Vault-5619: Transit BYOK Documentation

### PR DESCRIPTION
**Pull request by rculpepper**
Docs changes for Transit BYOK API endpoints and description for constructing the ciphertext for the key

### CODE REVIEW

File: website/content/api-docs/secret/transit.mdx [hunk 1]
```diff
@@ -91,6 +91,155 @@ $ curl \
...
+    --request POST \
+    --data @payload.json \
+    http://127.0.0.1:8200/v1/transit/keys/my-key/import
+```
+
+## Import Key Version
+
+This endpoint imports new key material into an existing imported key.
+
+**Note**: Keys whose material was generated by Vault do not support
```
Comment (taoism4504) [inline website/content/api-docs/secret/transit.mdx:179]:
```suggestion
~> **Note:** Keys whose material was generated by Vault do not support
```

File: website/content/api-docs/secret/transit.mdx [hunk 2]
```diff
@@ -266,6 +416,10 @@ ciphertext to be encrypted with the latest version of the key, use the `rewrap`
 endpoint. This is only supported with keys that support encryption and
 decryption operations.
 
+**Note**: For imported keys, rotation is only supported if the
```
Comment (taoism4504) [inline website/content/api-docs/secret/transit.mdx:419]:
```suggestion
~> **Note:** For imported keys, rotation is only supported if the
```

**Review by taoism4504 — APPROVED** [review]
Minor nits, but otherwise LG. Please ensure to specify the 1.11 label if these changes only apply to the upcoming Vault 1.11 and you don't want these changes published until the release. Otherwise, please set the label to backlog/1.10.x

File: website/content/api-docs/secret/transit.mdx [hunk 3]
```diff
@@ -91,6 +91,155 @@ $ curl \
...
+- `name` `(string: <required>)` – Specifies the name of the encryption key to
+  create. This is specified as part of the URL.
+
+- `ciphertext` `(string: <required>)` - A base64-encoded string that contains
+two values: an ephemeral AES key wrapped using the wrapping key
+returned by Vault and the encryption of the import key material under the
+provided AES key. The wrapped AES key should be the first 512 bytes of the
+ciphertext, and the encrypted key material should be the remaining bytes.
+
+- `hash_function` `(string: "SHA-256")` - The hash function used for the
```
Comment (stevendpclark) [inline website/content/api-docs/secret/transit.mdx:110]: Comparing against the code looks like we are expecting non-dashed versions of the hash function names such as 
`SHA256`...

File: website/content/api-docs/secret/transit.mdx [hunk 4]
```diff
@@ -91,6 +91,155 @@ $ curl \
     http://127.0.0.1:8200/v1/transit/keys/my-key
 ```
 
+## Import Key
+
+This endpoint imports existing key material into a new transit-managed encryption key.
+To import key material into an existing key, see the `import_version/` endpoint.
+
+### Parameters
```
Comment (stevendpclark) [inline website/content/api-docs/secret/transit.mdx:99]: I believe we are missing the [context](https://github.com/hashicorp/vault/blob/main/builtin/logical/transit/path_import.go#L77-L83) parameter.

**Review by stevendpclark — APPROVED** [review]

**Pull request merged**